### ISSUE TITLE
Clippy fails for "unused imports" when imports are used in rustdoc

### ISSUE THREAD

**Issue by hamza1311** [rust-lang/rust-clippy#6327]
Clippy fails for unused imports when imports are used in rustdoc. This feature is [documented](https://doc.rust-lang.org/beta/rustdoc/linking-to-items-by-name.html) and is available in beta and should be supported by clippy

I tried this code: https://github.com/hamza1311/clippy-bug-repro

### Expected results
clippy to succeed because `Second` is used in rustdoc

### Actual results
Fails with "error: unused import: `crate::second::Second`"  
Logs: https://github.com/hamza1311/clippy-bug-repro/runs/1390092464#step:5:7

There is an error in the logs instead of warning because I ran clippy with `-D warnings`.   
Workflow file: https://github.com/hamza1311/clippy-bug-repro/blob/master/.github/workflows/lint.yml#L32

### Meta

- `cargo clippy -V`:  `clippy 0.0.212 (5404efc 2020-11-11)`
- `rustc -Vv`:
```
rustc 1.49.0-nightly (5404efc28 2020-11-11)
binary: rustc
commit-hash: 5404efc28a0cddee103ef6396c48ea71ff9631c8
commit-date: 2020-11-11
host: x86_64-unknown-linux-gnu
release: 1.49.0-nightly
```

<details><summary>Backtrace</summary>
  <p>
  
  ```
  error: unused import: `crate::second::Second`
 --> src/first.rs:1:5
  |
1 | use crate::second::Second;
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-D unused-imports` implied by `-D warnings`
  ```
  
  </p>
</details>

**Comment by flip1995** [comment]
`unused_imports` is a rustc lint, not a Clippy lint. Is the `use` statement necessary for rustdoc to correctly link to `Second`?

**Comment by hamza1311** [comment]
> `unused_imports` is a rustc lint, not a Clippy lint
Huh. I assumed clippy controls that when it is invoked. Is this not the right place for this bug report? If so, where should I report it?

> Is the use statement necessary for rustdoc to correctly link to Second?
Yes, without it, the following error shows up

```rust
warning: unresolved link to `Second`
 --> src/first.rs:3:12
  |
3 | /// Using [`Second`]
  |            ^^^^^^^^ no item named `Second` in scope
  |
  = note: `#[warn(broken_intra_doc_links)]` on by default
  = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

**Comment by flip1995** [comment]
The `use` statement is necessary, if you want to write ``[`Second`]``. But you can also write ``[`Second`][`crate::second::Second`]`` which works without an import.

I don't think this is a bug. If you still consider this as a bug, please report it in rust-lang/rust or ask on the [`rustdoc` Discord](https://discord.gg/SpZ8R2wK).

> Huh. I assumed clippy controls that when it is invoked. Is this not the right place for this bug report? If so, where should I report it?

Nope, Clippy only adds lints to rustc. rustc itself also has lints. You know if it is a Clippy lint, if the lint name starts with `clippy::*`.
